packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -266,7 +266,7 @@ export class EngineReportingExtension<TContext = any>
     }
 
     const path = info.path;
-    const node = this.newNode(path);
+    const node = this.newNode(path, info);
```

Comment:
While this does seem like it would work, it seems slightly weird to have the presence/lack of an `info` parameter be the toggling factor, particularly in a recursively invoked method.

Maybe I'm just having a hard time groking it, but since `newNode` and `ensureParentNode` are both `private` methods, could we change the existing signatures of both of those to receive `info` (the object which contains `path`) directly rather than `path` as their only argument and use `info.fieldName === path.key` be the alias-determining-attribute?

Reply:
@abernix That would be better, but do we have access to the `info` to pass in for the parent? If I'm understanding correctly `willResolveField` only receives `GraphQLResolveInfo` for the current field